server/src/test/java/org/elasticsearch/TransportVersionTests.java

```diff
@@ -0,0 +1,110 @@
+        );
+        assertEquals(
+            TransportVersion.fromId(1_01_01_99),
+            TransportVersion.min(TransportVersion.fromId(1_01_01_99), TransportVersion.CURRENT)
+        );
+        TransportVersion version = TransportVersionUtils.randomVersion();
+        TransportVersion version1 = TransportVersionUtils.randomVersion();
+        if (version.id <= version1.id) {
+            assertEquals(version, TransportVersion.min(version1, version));
+        } else {
+            assertEquals(version1, TransportVersion.min(version1, version));
+        }
+    }
+
+    public void testMax() {
+        assertEquals(TransportVersion.CURRENT, TransportVersion.max(TransportVersion.CURRENT, TransportVersionUtils.getPreviousVersion()));
+        assertEquals(TransportVersion.CURRENT, TransportVersion.max(TransportVersion.fromId(1_01_01_99), TransportVersion.CURRENT));
+        TransportVersion version = TransportVersionUtils.randomVersion();
+        TransportVersion version1 = TransportVersionUtils.randomVersion();
+        if (version.id >= version1.id) {
+            assertEquals(version, TransportVersion.max(version1, version));
+        } else {
+            assertEquals(version1, TransportVersion.max(version1, version));
+        }
+    }
+
+    public void testVersionConstantPresent() {
+        // TODO those versions are not cached at the moment, perhaps we should add them to idToVersion set too?
+        Set<TransportVersion> ignore = Set.of(TransportVersion.ZERO, TransportVersion.CURRENT, TransportVersion.MINIMUM_COMPATIBLE);
+        assertThat(TransportVersion.CURRENT, sameInstance(TransportVersion.fromId(TransportVersion.CURRENT.id)));
+        final int iters = scaledRandomIntBetween(20, 100);
+        for (int i = 0; i < iters; i++) {
+            TransportVersion version = TransportVersionUtils.randomVersion(ignore);
+
+            assertThat(version, sameInstance(TransportVersion.fromId(version.id)));
+        }
+    }
+
+    public void testCURRENTIsLatest() {
+        final int iters = scaledRandomIntBetween(100, 1000);
+        for (int i = 0; i < iters; i++) {
+            TransportVersion version = TransportVersionUtils.randomVersion();
+            if (version != TransportVersion.CURRENT) {
+                assertThat(
+                    "Version: " + version + " should be before: " + Version.CURRENT + " but wasn't",
+                    version.before(TransportVersion.CURRENT),
+                    is(true)
+                );
+            }
+        }
+    }
+
+    public void testToString() {
+        assertEquals("5000099", TransportVersion.fromId(5_00_00_99).toString());
+        assertEquals("2030099", TransportVersion.fromId(2_03_00_99).toString());
+        assertEquals("1000099", TransportVersion.fromId(1_00_00_99).toString());
+        assertEquals("2000099", TransportVersion.fromId(2_00_00_99).toString());
+        assertEquals("5000099", TransportVersion.fromId(5_00_00_99).toString());
+    }
+}
```

Comment:
I think we are missing a comprehensive test of all the version constants. See VersionTests.testAllVersionsMatchId()

Reply:
Added test